### PR TITLE
Bellis use owner id for owner

### PR DESCRIPTION
**Pull request by ellisbenjamin**
As a user of namely-connect I should be able to change the owner of the installation via the view all users page.

https://namely.atlassian.net/browse/CORE-302

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
 class UsersController < ApplicationController
   def index
+    @installation_id = current_user.installation.id
     @users = current_user.installation.users
+    @owner_id = current_user.installation.owner.id
+  end
+
+  def update
+    current_user.installation.update(owner_id: params[:user_radio])
```
Comment (bobbytables) [inline app/controllers/users_controller.rb]: There's a small security concern here. There's a possibility of changing the installations owner_id to an ID that someone has posed as.

Proposed solution:

Find the User in the database for the installation before updating this attribute. So:

```ruby
def update
  installation = current_user.installation
  if user = installation.users.find(params[:user_radio])
    current_user.installation.update(owner_id: user.id)
  end
end
```

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+      <tr>
+        <th>Owner?</th>
+        <th>First name</th>
+        <th>Last name</th>
+        <th>Email</th>
+      </tr>
+        <% @users.each do |user| %>
+          <tr>
+              <td><%= radio_button_tag("user_radio", user.id, user.id == @owner_id) %>
+              </td>
```
Comment (bobbytables) [inline app/views/users/index.html.erb]: Formatting.

File: spec/features/user_views_list_of_users_and_owners_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    expect(find("#user_radio_#{users[0].id}")).to_not be_checked
+    expect(find("#user_radio_#{users[1].id}")).to be_checked
+    expect(find("#user_radio_#{users[2].id}")).to_not be_checked
+  end
+
+
+
+
+end
```
Comment (bobbytables) [inline spec/features/user_views_list_of_users_and_owners_spec.rb]: Formatting :)

File: spec/features/user_views_list_of_users_and_owners_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    click_button "Save"
+
+    visit dashboard_path(as: users[0])
+    click_link t("dashboards.show.view_all_users")
+
+    expect(find("#user_radio_#{users[0].id}")).to_not be_checked
+    expect(find("#user_radio_#{users[1].id}")).to be_checked
+    expect(find("#user_radio_#{users[2].id}")).to_not be_checked
+  end
```
Comment (bobbytables) [inline spec/features/user_views_list_of_users_and_owners_spec.rb]: This scenario is great!

**Comment by bobbytables** [comment]
Minus the one comment this looks great :)

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (ellisbenjamin) [inline app/controllers/users_controller.rb]: Thanks for the catch.  Implemented as:

```
  def update
    installation = current_user.installation
    if user = installation.users.find(params[:user_radio])
      installation.update(owner_id: user.id)
      redirect_to users_path
    else
      redirect_to users_path
    end
  end
```

File: spec/features/user_views_list_of_users_and_owners_spec.rb (hunk 3 above)
Comment (ellisbenjamin) [inline spec/features/user_views_list_of_users_and_owners_spec.rb]: woops.  fixed.

File: app/views/users/index.html.erb (hunk 2 above)
Comment (ellisbenjamin) [inline app/views/users/index.html.erb]: Moved that td tag from line 15 to 14

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+    @installation_id = current_user.installation.id
     @users = current_user.installation.users
+    @owner_id = current_user.installation.owner.id
+  end
+
+  def update
+    installation = current_user.installation
+    if user = installation.users.find(params[:user_radio])
+      installation.update(owner_id: user.id)
+      redirect_to users_path
```
Comment (cored) [inline app/controllers/users_controller.rb]: @ellisbenjamin Looks like you always redirecting, why not move that line into the button and remove the else? Also you could push this logic into the model and make it like. 
Also looks like that what you doing with the data is, an update of the owner, what do you think about having a method call like: `installation.update_owner(params[:user_radio])` ?

**Comment by cored** [comment]
:+1:

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (bobbytables) [inline app/controllers/users_controller.rb]: `update_owner` is a good idea.

Comment (ellisbenjamin) [inline app/controllers/users_controller.rb]: thanks @cored!

I gave the Installation model an update_owner method and updated the redirect logic.  We are always redirecting back to the users_path so I updated that little bit of logic too.  I'm not sure that I understand moving the redirect to the button

```
  def update_owner(owner_id)
    update_attribute(:owner_id, owner_id)
  end
```

**Pull request merged**